Cognite.Simulator.Extensions/SequencesExtensions.cs

```diff
@@ -566,6 +567,106 @@ public static string[] GetStringValues(this SequenceRow row)
+                {
+                    RowNumber = r.RowNumber,
+                    Values = r.Values
+                }).ToList();
+            // Checking to see if the key alread exists
+            var foundIndex = newRows.FindIndex(0, newRows.Count, (item => item.Values.FirstOrDefault().ToString().Contains(updateKey) ));
+            if (foundIndex == -1){
+                newRows.Insert(newRows.Count , new SequenceRow{
+                    RowNumber = newRows.Count,
+                    Values = new List<MultiValue> {
+                        MultiValue.Create(updateKey),
+                        MultiValue.Create(updateValue)
+                    }
+                });
+            } else {
+                newRows[foundIndex].Values = new List<MultiValue> {
+                    MultiValue.Create(updateKey),
+                    MultiValue.Create(updateValue)
+                };
+            }
+
+            SequenceDataCreate sequenceData = new SequenceDataCreate
+            {
+                ExternalId = sequenceExternalId,
+                Columns = new List<string> {
+                            KeyValuePairSequenceColumns.Key,
+                            KeyValuePairSequenceColumns.Value
+                        },
+                Rows = newRows
+            };
+            var rowsToCreate = new List<SequenceDataCreate>();
+                rowsToCreate.Add(sequenceData);
+                var result = await sequences.InsertAsync(
+                    rowsToCreate,
+                    keyChunkSize: 10,
+                    valueChunkSize: 100,
+                    sequencesChunk: 10,
+                    throttleSize: 1,
+                    RetryMode.None,
+                    SanitationMode.None,
+                    token
+                ).ConfigureAwait(false);
+                
+            if (!result.IsAllGood)
+            {
+                throw new SimulatorIntegrationSequenceException("Could not update simulator integration sequence", result.Errors);
+            }
+        }
+
+        /// <summary>
+        /// Gets the externalId of a SimulatorIntegrations sequence (This function can throw an exception)
+        /// </summary>
+        /// <param name="sequences">CDF Sequence resource</param>
+        /// <param name="simulatorName">The simulator name.</param>
+        /// <param name="simulatorDataSetId">The simulator datasetId</param>
+        /// <param name="connectorName">The connector Name</param>
+        /// <param name="token">The cancellation token</param>
+        /// <returns>externalId string</returns>
+        public static async Task<string> GetSequenceExternalId(this SequencesResource sequences, string simulatorName, long simulatorDataSetId, string connectorName, CancellationToken token) {
+             var _simulatorSequenceIds = new Dictionary<string, string>();
+            List<string> _simulators = new List<string> { simulatorName };
+            var simulatorsDict = _simulators.Select(
```

Comment:
Since it is only one item (`simulationName`), this `Select` is not necessary, and the `_simulators` list is also unnecessary